### DISCUSSION TITLE
Add keyboard keys formatted in markdown while user does not type EOF

### DISCUSSION THREAD

**Discussion by Blackstareye** [SilentVoid13/Templater#1403] [Templates Showcase]
## Instruction 

   * press <kbd>Alt</kbd>+<kbd>E</kbd> and select this template via the name you gave it to open its modal dialog
 
   * spell out the first key you'd like to "keyboardify", e.g. `a`, `ctrl`, `del` etc. IN PLAIN TEXT (so no need to press the <kbd>Ctrl</kbd> or <kbd>Alt</kbd> key etc. themselves, this will do nothing)

   * press enter: this will record the entered text/key and clear the prompt

   * enter the second key in the same way

   * (rinse and repeat until done)

   * type EOF in an empty prompt to exit and insert the complete sequence converted to the proper markdown/html sequence.
 
instruction  improved by: [neitsab](https://github.com/neitsab)  - thx ^.^
<hr>
Example: 
   User-Input 1: a
   User-Input 2: ctrl
   User-Input 3: space
   User-Input 4: EOF

Output: <kbd>a</kbd>+<kbd>ctrl</kbd>+<kbd>space</kbd>

```
<%*
  var container = [];
  var response = "";
  
  do {
	  response = await tp.system.prompt("Type your Key. If you are finished type: 'EOF'", null, false);
	  if (response == null) {
		  break;
	  }
	  if (response == "") {
		  continue;
	  }
	  if (response != "EOF") {
		  container.push("<kbd>"+response+"</kbd>");
	  }
	  
  } while (response != "EOF");
  
  
%>
<% container.join("+") %>
```
Edit: updated code, cause @neitsab  found a `PTSD-Vi`-like bug in which the dialog can't be quitted until you type `EOF` :'D

Now it can be canceled with esc or clicking the button

Gist: https://gist.github.com/Blackstareye/a73166ec17194e4d2a12c84edaedfbcf

**Comment by neitsab** [comment]
Hey, after being very confused while trying it, here's my reformulation of the instructions:

* press <kbd>Alt</kbd>+<kbd>E</kbd> and select this template via the name you gave it to open its modal dialog
* spell out the first key you'd like to "keyboardify", e.g. `a`, `ctrl`, `del` etc. IN PLAIN TEXT (so no need to press the <kbd>Ctrl</kbd> or <kbd>Alt</kbd> key etc. themselves, this will do nothing)
* press enter: this will record the entered text/key and clear the prompt
* enter the second key in the same way
* (rinse and repeat until done)
* type EOF in an empty prompt to exit and insert the complete sequence converted to the proper markdown/html sequence.

So quite interesting, but also quite jarring.

Btw, I couldn't exit by pressing the "close" button on the modal, so this brought up some "first time Vi" PTSD...  I was about to force close Obsidian :smile:

**Reply by Blackstareye** [reply]
Ohhhhh that is much better, thx! :D I will edit it and mention you  if that is okay for you :) 

>Btw, I couldn't exit by pressing the "close" button on the modal, so this brought up some "first time Vi" PTSD... I was about to force close Obsidian 😄

I didn't think about that case, sorry about that :'D 
PS: You just had to type EOF and everything would be as expected ^^'

**Reply by neitsab** [reply]
No worries I did eventually find it without having to burn my computer down :yum: 

That would be great, thanks a lot for your quick reply and sharing this script!

**Reply by Blackstareye** [reply]
If you got time, can you check if this snippet now works ? 
On my instance it works like it should, but 4 eyes are better than 2 :D  
@neitsab 

and you are welcome :)

**Reply by neitsab** [reply]
I can confirm that now the modal does close as expected on my system (Archlinux + Obsidian 1.6.7-1 from the official repos for the record)!

Thanks 🙏🏻
